framework_tvb/tvb/interfaces/web/controllers/spatial/local_connectivity_controller.py

```diff
@@ -326,6 +327,8 @@ def get_equation_chart(self):
             current_lconn = common.get_from_session(KEY_LCONN)
             surface_gid = current_lconn.surface.hex
             surface = ABCAdapter.load_entity_by_gid(surface_gid)
+            if surface is None:
```

Comment:
With my above suggestion then this check will change into `if current_lconn.surface is None:` and should be move upper.

Reply:
because if we do not remove the new_lconn.surface = uuid.uuid4() from the above comment he still have chere current_lconn.surface, and the surface is the one that is None